docs/drafts/initial-upgrade-support.md

```diff
@@ -0,0 +1,43 @@
+# Initial upgrade support
+
+This document explains how OLM 1.0 handles upgrades.
+
+OLM 1.0 introduces initial upgrade support with simplified UX for package authors and package admins which allows defining upgrade edges implicitly via [Semantic Versioning](https://semver.org/) standard.
+
+It also introduces an API to enable independently verified upgrades and downgrades.
+
+## Upgrade constraint semantics
+
+OLM 1.0 at the moment supports two upgrade constraint semantics:
+
+* [Semantic Versioning](https://semver.org/) (Semver)
+* [Legacy OLM 0 semantics](https://olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/#methods-for-specifying-updates)
+
+**Note:** At the moment of writing legacy semantics support in OLM 1.0 is incomplete: only `replaces` directive is taken into account.
+
+Kubernetes manifests provided in this repo enable Semver support by default, but cluster admins can control which semantics to use by passing one of the arguments into `manager` binary:
+* `--feature-gates=ForceSemverUpgradeConstraints=true` - enable Semver
+* `--feature-gates=ForceSemverUpgradeConstraints=false` - disable Semver, use legacy semantics
+
```

Comment:
Added an example calls. Or should we include example manifests?

Reply:
I think including example manifests would be helpful. Because you can set the argument using either the CLI or by editing the manifest/yaml file?

And sorry for not mentioning it yesterday, but I am not familiar with the `manager` binary. Is that something new to the operator-controller or is it a third party tool? Do we need to introduce how to download the binary or include it as a prerequisite?

Reply:
@michaelryanpeter the `manager` binary is operator-controller itself. This repo contains this manifest:

https://github.com/operator-framework/operator-controller/blob/6cb7029801558cd9e69fd4285468e600f45df7a6/config/manager/manager.yaml#L68-L72

But we later customise args like so:

https://github.com/operator-framework/operator-controller/blob/6cb7029801558cd9e69fd4285468e600f45df7a6/config/default/manager_auth_proxy_patch.yaml#L48-L53

Do you think we should have a full `Deployment` example for both states of the flag? Maybe under a spoiler? Or should we include only a part of the deployment manifest? E.g.

```yaml
- command:
  - /manager
  args:
  - --feature-gates=ForceSemverUpgradeConstraints=true
  image: controller:latest
```

Reply:
I think it would be work explaining. I read the instructions like I needed to enter a CLI command to enable the command, something like `kubectl manager --feature-gates=<etc>`.

I think that at this stage of development, just an excerpt like your last example would work. Then when we stitch the version, version range, channel, and upgrade docs together, we can maybe add another formulary of real world example manifests.